src/Lean/PrettyPrinter/Delaborator/Builtins.lean

```diff
@@ -24,13 +24,13 @@ def unfoldMDatas : Expr → Expr
 
 @[builtin_delab fvar]
 def delabFVar : Delab := do
-let Expr.fvar fvarId ← getExpr | unreachable!
-try
-  let l ← fvarId.getDecl
-  maybeAddBlockImplicit (mkIdent l.userName)
-catch _ =>
-  -- loose free variable, use internal name
-  maybeAddBlockImplicit <| mkIdent fvarId.name
+  let Expr.fvar fvarId ← getExpr | unreachable!
+  try
+    let l ← fvarId.getDecl
+    maybeAddBlockImplicit (mkIdent l.userName)
+  catch _ =>
+    -- loose free variable, use internal name
+    maybeAddBlockImplicit <| mkIdent (fvarId.name.replacePrefix `_uniq `_fvar)
```

Comment:
Would it make sense to generate the name as `_fvar` in the first place? Or are there unique names around that aren't free variables?

Reply:
I looked into that briefly while creating the PR, and `_uniq` is coming from the name generator configuration. That generator is used directly by `Lean.mkFreshFVarId` and `Lean.mkFreshMVarId`. The `mvar` delaborator also replaces `_uniq` like this.

It would be nice if the generated names could say `fvar` and `mvar` for debugging purposes.

Reply:
ok, if there is precedent for this replacement idiom, then all is well